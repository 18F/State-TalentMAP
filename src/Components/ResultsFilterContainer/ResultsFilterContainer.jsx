import React from 'react';
import PropTypes from 'prop-types';
import { FILTER_ITEMS_ARRAY, ACCORDION_SELECTION_OBJECT } from '../../Constants/PropTypes';
import { ACCORDION_SELECTION } from '../../Constants/DefaultProps';
import SearchFiltersContainer from '../SearchFilters/SearchFiltersContainer/SearchFiltersContainer';
import ResetFilters from '../ResetFilters/ResetFilters';

const ResultsFilterContainer = ({ filters, onQueryParamUpdate, onChildToggle,
  onQueryParamToggle, resetFilters, setAccordion, selectedAccordion }) => (
    <div className="filter-container">
      <div className="filter-container-bottom">
        <div className="filter-control-container">
          <div className="filter-control-left">Select Filter</div>
          <div className="filter-control-right">
            <ResetFilters resetFilters={resetFilters} />
          </div>
        </div>
        <br />
        <div className="usa-grid-full search-filters-container">
          <SearchFiltersContainer
            filters={filters}
            queryParamUpdate={(e) => {
              onQueryParamUpdate(e);
              onChildToggle();
            }}
            queryParamToggle={(param, value, remove) => {
              onQueryParamToggle(param, value, remove);
              onChildToggle();
            }}
            selectedAccordion={selectedAccordion}
            setAccordion={setAccordion}
          />
        </div>
      </div>
    </div>
);

ResultsFilterContainer.propTypes = {
  filters: FILTER_ITEMS_ARRAY.isRequired,
  onQueryParamUpdate: PropTypes.func.isRequired,
  onChildToggle: PropTypes.func.isRequired,
  onQueryParamToggle: PropTypes.func.isRequired,
  resetFilters: PropTypes.func.isRequired,
  setAccordion: PropTypes.func.isRequired,
  selectedAccordion: ACCORDION_SELECTION_OBJECT,
};

ResultsFilterContainer.defaultProps = {
  selectedAccordion: ACCORDION_SELECTION,
};

export default ResultsFilterContainer;
