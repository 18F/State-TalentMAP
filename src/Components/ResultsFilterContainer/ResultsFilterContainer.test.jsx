import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import ResultsFilterContainer from './ResultsFilterContainer';

describe('ResultsFilterContainerComponent', () => {
  const items = [{
    title: 'title', expanded: true,
  }];

  it('is defined', () => {
    const wrapper = shallow(
      <ResultsFilterContainer
        filters={items}
        onQueryParamUpdate={() => {}}
        onChildToggle={() => {}}
        onQueryParamToggle={() => {}}
        resetFilters={() => {}}
        setAccordion={() => {}}
        selectedAccordion=""
      />);
    expect(wrapper).toBeDefined();
  });

  it('can receive props', () => {
    const wrapper = shallow(<ResultsFilterContainer
      filters={items}
      onQueryParamUpdate={() => {}}
      onChildToggle={() => {}}
      onQueryParamToggle={() => {}}
      resetFilters={() => {}}
      setAccordion={() => {}}
      selectedAccordion=""
    />);
    expect(wrapper.instance().props.filters).toBe(items);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<ResultsFilterContainer
      filters={items}
      onQueryParamUpdate={() => {}}
      onChildToggle={() => {}}
      onQueryParamToggle={() => {}}
      resetFilters={() => {}}
      setAccordion={() => {}}
      selectedAccordion=""
    />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
