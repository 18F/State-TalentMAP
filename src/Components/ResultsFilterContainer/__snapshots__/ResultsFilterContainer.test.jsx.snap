// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsFilterContainerComponent matches snapshot 1`] = `
<div
  className="filter-container"
>
  <div
    className="usa-grid-full filter-container-top"
  >
    Keywords
  </div>
  <div
    className="usa-grid-full filter-container-search"
  >
    <SearchBar
      alertText="Disabled"
      defaultValue={null}
      id="searchbar-filter-container"
      label="Search"
      labelSrOnly={true}
      noButton={false}
      noForm={false}
      onChangeText={[Function]}
      onSubmitSearch={[Function]}
      placeholder="Position, Region, Posts"
      submitDisabled={false}
      submitText="search"
      type="small"
    />
  </div>
  <div
    className="filter-container-bottom"
  >
    <div
      className="filter-control-container"
    >
      <div
        className="filter-control-left"
      >
        Select Filter
      </div>
      <div
        className="filter-control-right"
      >
        <ResetFilters
          resetFilters={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="usa-grid-full search-filters-container"
    >
      <SearchFiltersContainer
        filters={
          Array [
            Object {
              "expanded": true,
              "title": "title",
            },
          ]
        }
        queryParamToggle={[Function]}
        queryParamUpdate={[Function]}
        selectedAccordion=""
        setAccordion={[Function]}
      />
    </div>
  </div>
</div>
`;
